shared/middleware/files_api.rb

```diff
@@ -337,6 +337,9 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
       quota_crossed_half_used(endpoint, encrypted_channel_id) if quota_crossed_half_used?(app_size, body.length)
     end
 
+    # Block libraries with PII/profanity from being published.
+    return bad_request if endpoint == 'libraries' && ShareFiltering.find_failure(body, request.locale)
```

Comment:
:+1: This seems like a good place to do this check.  I did notice that calling `ShareFiltering.find_failure` directly bypasses the gatekeeper checks in that module; I wonder if it's worth a little refactoring in there to make sure our gatekeeper config still allows us to shut down filtering checks if we need to.

Reply:
ooo, good catch. adding that check to `ShareFiltering.find_failure`